app/services/create_vote_service.rb

```diff
@@ -0,0 +1,50 @@
+class CreateVoteService
+
+  def initialize(votable,current_user,params)
+    @votable = votable
+    @current_user = current_user
+    @params = params
+  end
+
+  def call
+    if @votable.votes.where(user_id: @current_user.id).present?
+      vote = @votable.votes.where(user_id: @current_user.id).first
+      if vote.value == 1
+        if (@params[:value] == "+1")
```

Comment:
looks like this is used in a couple places, could probably make some nicely named methods for it, `upvoting?` and `downvoting?` perhaps?